resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/SparkKubernetesClientFactory.scala

```diff
@@ -42,6 +42,9 @@ private[spark] object SparkKubernetesClientFactory {
       sparkConf: SparkConf,
       defaultServiceAccountToken: Option[File],
       defaultServiceAccountCaCert: Option[File]): KubernetesClient = {
+
+    // TODO Support configurable context
+
```

Comment:
Note to reviewers.  In testing this I noticed one issue is that when you have multiple contexts defined in your `KUBECONFIG` file that while this PR adds the ability for the integration test machinery to use an arbitrary context the submission client will always use your current context.  Therefore if your current context does not match the requested context the integration tests will fail/return false positives because the client and the test harness will use different contexts.

I have filed SPARK-25887 for addressing this since the client change seemed like it should be a separate issue and PR.  I will provide a PR for that tomorrow.